### ISSUE TITLE
[Network] Insane number of DNS requests originating from fresh Rockstor VM

### ISSUE THREAD

**Issue by Overwatching** [rockstor/rockstor-core#2291]
I have a freshly installed Rockstor 3.9.1-0 VM in Proxmox.

The next day I checked my Pi-hole dashboard and noticed some...odd behavior.

That VM generated ~200k DNS requests in the last 16 hours.

The requests are not being blocked. They are resolving properly, though 99% of them are hitting the DNS cache (obviously since the same addresses are being hit every other second.)

The Pi-hole logs for that machine are so large that I had to cut it into chunks and grab them manually. (So I have a Pi-hole bug report to post as well, since it seems possible to crash the web-renderer by viewing an oversized log...)

This is the top 10 domains being resolved in the last 8 hours and the number of requests

- mirror.sfo12.us.leaseweb.net | 15287 |  
- mirror.lax.genesisadaptive.com | 11328 |  
- repo1.dal.innoscale.net | 11283 |  
- mirror.den01.meanservers.net | 11273 |  
- mirror.chpc.utah.edu | 10693 |  
- mirrors.xmission.com | 10668 |  
- dfw.mirror.rackspace.com | 10664 |  
- mirrors.lug.mtu.edu | 10647 |  
- mirror.dal.nexril.net | 10647 |  
- packages.oit.ncsu.edu | 10646 |  

These look like package mirrors maybe? But I cannot imagine what might cause this kind of behavior.

**Comment by phillxnet** [comment]
@Overwatching Thanks for the report. 3.9.1-0 is super old but our last pre-compiled installer. You are likely to have a very different experience with our more modern offering that is only currently available via the following DIY installer build recipe:

https://github.com/rockstor/rockstor-installer

**3.9.1-0 installer was released 2nd July 2017:**

https://github.com/rockstor/rockstor-core/releases/tag/3.9.1

> These look like package mirrors maybe?

Yes that is my guess too.

Plus from your second 'suspect' entry:
http://mirror.lax.genesisadaptive.com/
we have it showing as a CentOS / Epel / Fedora mirror.

 At one point our yum config refreshed every minute. This is obviously excessive and was addressed in:

"add non legacy distro aware repo configuration. Fixes #1991" #1992

https://github.com/rockstor/rockstor-core/releases/tag/3.9.2-47

And this was released in the Stable Branch on 22 Feb 2019

And as we move to using both yum (dnf-yum actually) and zypper in our new 'Built on openSUSE' variant we still end up doubling up our package look-ups but given the yum config was moved from 1 minute to 1 hour (in above referenced fix) and zypper was left at it's default I think we should be better behaved now.

Latest testing release in 4 'Built on openSUSE' variant was 4.0.6:
https://forum.rockstor.com/t/beta-built-on-opensuse-testing-channel-changelog/7097/13
8th March 2021

And we have a placeholder rpm of 4.0.6 in the Stable channel until we do our next official Stable release.

Let us know if you see the same behaviour in anything after 3.9.2-47 (Stable channel in CentOS, or either channel in the 'Built on openSUSE' 4 variant.

Also is your instance without rock-ons?

Hope that helps to narrow down what may be happening here.

**Comment by Overwatching** [comment]
1. No rock-ons. It was a fresh install. The only thing I did beyond the installer was to add a drive and make a SMB share for testing purposes.
2. This was for my HomeLab, which I'm spinning up to give me a place to actually have a reason to learn Linux stuff. But at the moment I'm only (barely) comfortable with the debian ecosystem. I'm not at the point where I'm ready to build anything myself at the moment. I'm at the "Take something working, tweak the knobs and see what happens." stage. I don't yet have a feel for any of the "why"s.
3. 3. I used the most recent .iso release because it was there. I understand so little of what I'm reading on https://github.com/rockstor/rockstor-installer that I'm honestly not even going to try. I don't have the time to learn everything from scratch right now.

Question though. Is there no upgrade path for Rockstor? Or is the last .iso build just so old that it cannot be upgraded? The first thing I've been doing after installing something is updating it, but a quick search shows that running something like "yum update" results in everything breaking and a re-install. When I saw that, I assumed that there was some other built in upgrade system. Or maybe even some UI alert that notified you of an update and the appropriate command-line to do it. Is that not the case?

**Comment by phillxnet** [comment]
> Question though. Is there no upgrade path for Rockstor?

One can save config and download and then import the pool and restore the config between an older 3 and a newer 4 variant:
http://rockstor.com/docs/disks.html#import-btrfs-pool
and:
http://rockstor.com/docs/config_backup/config_backup.html

No direct update as one is based on CentOS and the other (4) is 'Built on openSUSE'. But the data pool format and rockstor code is compatible re config save / restore. Although obviously elements added to the config save/restore after the last save will not be in the config save. We have for instance more of late added the ability to restore Rock-ons (within some limitations) if the config was taken before this feature was added (4 only) then it will not be in the config (ie it wont work) when you do a restore because it's just not in the config save at the time. But all the rest should work as we have kept the 'system' compatible.

Hope that helps.

These types of questions are best addressed to the forum really:
https://forum.rockstor.com/
